docs/development/internal_api.rst

```diff
@@ -12,3 +20,73 @@ Internal API
    :undoc-members:
    :private-members:
    :show-inheritance:
+
+Extensions
+----------
+.. automodule:: blocks.extensions.predicates
+   :undoc-members:
+   :members:
+   :private-members:
+   :show-inheritance:
+
+Monitoring
+----------
+.. automodule:: blocks.monitoring.aggregation
+   :undoc-members:
+   :members:
+   :private-members:
+   :show-inheritance:
+
+.. automodule:: blocks.monitoring.evaluators
+   :undoc-members:
+   :members:
+   :private-members:
+   :show-inheritance:
+
+Search
+------
+.. automodule:: blocks.search
```

Comment:
Some of these modules shouldn't go into the internal API. The idea is that the internal API is used to document modules and objects that a user would normally not interface with (but a developer would) e.g. `ApplicationCall`, `BoundApplication`, that sort of things. Beam search, the wrapper bricks, and some of the methods in the aggregation module should be public though.